csrc/contiguity.cpp

```diff
@@ -527,7 +527,7 @@ void ContigIDs::build(const std::vector<IterDomain*>& ids) {
     // rfactor root domains, which should just return "zero"
     // RootAxisInfo. This should be safe as no rfactor tensor should
     // need halo.
-    if (*root_contiguity_.at(root_domain_i) &&
+    if (root_contiguity_.at(root_domain_i).value_or(false) &&
```

Comment:
I think you should `TORCH_INTERNAL_ASSERT(root_contiguity_.at(root_domain_i).has_value())`. Having no value mean doesn't mean it is discontiguous, but instead, it means a bug.

Reply:
explicit assert sounds good~ The bar is that we don't want to change behavior just for format changes. is calling `value()` directly triggering any clang-tidy error?

Reply:
The clang-tidy error is:
```
  Error (CLANGTIDY) [bugprone-unchecked-optional-access,-warnings-as-errors]
    unchecked access to optional value
```

We need to acquire the optional value before checking it.